presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -105,9 +105,13 @@
     private final Deque<Iterator<InternalHiveSplit>> fileIterators = new ConcurrentLinkedDeque<>();
 
     // Purpose of this lock:
+    // * For operations that require exclusive access to either: partitions, fileIterators or hiveSplitSource
+    //   use a write lock
+    // * For checking state use a read lock
+    // Details:
     // * When write lock is acquired, except the holder, no one can do any of the following:
-    // ** poll from partitions
-    // ** poll from or push to fileIterators
+    // ** poll from (or check empty) partitions
+    // ** poll from or push to (or check empty) fileIterators
```

Comment:
poll from (or check empty) or push to fileIterators

Reply:
Updated.